AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DatabaseErrorDialog.java

```diff
@@ -223,7 +223,16 @@ public MaterialDialog onCreateDialog(Bundle savedInstanceState) {
                                         return true;
                                     });
                 }
-                return builder.show();
+                MaterialDialog materialDialog = builder.build();
+                materialDialog.setOnKeyListener((dialog, keyCode, event) -> {
+                    if (keyCode == KeyEvent.KEYCODE_BACK) {
+                        Timber.i("catch");
```

Comment:
Nice to have: Make the log a little more expressive 
```suggestion
                        Timber.i("Back Pressed - Closing 'Restore From Backup'");
```